solr/core/src/test/org/apache/solr/cloud/TestPullReplica.java

```diff
@@ -191,91 +270,56 @@ public void testCreateDelete() throws Exception {
     }
   }
 
-  /**
-   * Asserts that Update logs don't exist for replicas of type {@link org.apache.solr.common.cloud.Replica.Type#PULL}
-   */
-  private void assertUlogPresence(DocCollection collection) {
-    for (Slice s:collection.getSlices()) {
-      for (Replica r:s.getReplicas()) {
-        if (r.getType() == Replica.Type.NRT) {
-          continue;
-        }
-        SolrCore core = null;
-        try {
-          core = cluster.getReplicaJetty(r).getCoreContainer().getCore(r.getCoreName());
-          assertNotNull(core);
-          assertFalse("Update log should not exist for replicas of type Passive but file is present: " + core.getUlogDir(),
-              new java.io.File(core.getUlogDir()).exists());
-        } finally {
-          core.close();
-        }
-      }
-    }
-  }
-
   @SuppressWarnings("unchecked")
-  // 12-Jun-2018 @BadApple(bugUrl="https://issues.apache.org/jira/browse/SOLR-12028")
   public void testAddDocs() throws Exception {
-    int numPullReplicas = 1 + random().nextInt(3);
+    int numPullReplicas = 1; // + random().nextInt(3);
     CollectionAdminRequest.createCollection(collectionName, "conf", 1, 1, 0, numPullReplicas)
-    .process(cluster.getSolrClient());
-    waitForState("Expected collection to be created with 1 shard and " + (numPullReplicas + 1) + " replicas", collectionName, clusterShape(1, numPullReplicas + 1));
+        .process(cluster.getSolrClient());
+    waitForState("Expected collection to be created with 1 shard and " + (numPullReplicas + 1) + " replicas",
+        collectionName, clusterShape(1, numPullReplicas + 1));
     DocCollection docCollection = assertNumberOfReplicas(1, 0, numPullReplicas, false, true);
     assertEquals(1, docCollection.getSlices().size());
 
-    boolean reloaded = false;
     int numDocs = 0;
-    while (true) {
+    CloudSolrClient solrClient = cluster.getSolrClient();
+    for (int i = 0; i < 5; i++) {
       numDocs++;
-      cluster.getSolrClient().add(collectionName, new SolrInputDocument("id", String.valueOf(numDocs), "foo", "bar"));
-      cluster.getSolrClient().commit(collectionName);
+
+      UpdateRequest ureq = new UpdateRequest();
+      ureq.add(new SolrInputDocument("id", String.valueOf(numDocs), "foo", "bar"));
+      ureq.process(solrClient, collectionName);
+      ureq.commit(solrClient, collectionName);
```

Comment:
just curious why this needed to be split out into two calls? if it's just for debugging, or there's some condition under test. leave a comment either way.